pkgs/native_assets_builder/pubspec.yaml

```diff
@@ -7,7 +7,9 @@ repository: https://github.com/dart-lang/native/tree/main/pkgs/native_assets_bui
 publish_to: none
 
 environment:
-  sdk: '>=3.3.0 <4.0.0'
+  # Allow any pre-release version of 3.5.0 and onwards
+  # (i.e. 3.5.0-edge*, 3.5.0-dev*, 3.5.0-beta*, ...)
+  sdk: '>=3.5.0-0 <4.0.0'
```

Comment:
I think this should be `'>=3.5.0 <4.0.0'` any dev/master commits after the branch cut of 3.5 should be `3.6.0-0`.

Also I just checked by downloading 3.5.0 from the Dart SDK archive, we can also delete support for v1.2 serialization, 3.5.0 has v1.3.

Reply:
Done - made it `>=3.5.0` (the `-0` would allow 3 months more - the master/dev/beta leading up to 3.5.0 - but I guess it's not necesary).

> Also I just checked by downloading 3.5.0 from the Dart SDK archive, we can also delete support for v1.2 serialization, 3.5.0 has v1.3.

There's a little bit more code that can be removed. Done.